server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1123,12 +1123,24 @@ public SnapshotInfo takeSnapshot(VolumeInfo volume) throws ResourceAllocationExc
             }
 
             SnapshotInfo snapshotOnPrimary = snapshotStrategy.takeSnapshot(snapshot);
-            if (payload.getAsyncBackup()) {
-                backupSnapshotExecutor.schedule(new BackupSnapshotTask(snapshotOnPrimary, snapshotBackupRetries - 1, snapshotStrategy), 0, TimeUnit.SECONDS);
+            boolean backupFlag = BackupSnapshotAfterTakingSnapshot.value() == null ||
+                    BackupSnapshotAfterTakingSnapshot.value();
+
+            if (backupFlag) {
+                if (payload.getAsyncBackup()) {
```

Comment:
can you extract lines 1130-1136 to a method? This will also allow you to document it, and then to unit test it.

Reply:
This isn't my code, but I'm not entirely sure how to even test things in backupSnapshotExecutor.  That is inited by the config() method, and that's not called from the existing unit tests.  Any thoughts?